### DISCUSSION TITLE
coin_balance_catchup errors, while block_catchup working fine

### DISCUSSION THREAD

**Discussion by ixje** [blockscout/blockscout#11497] [Support]
In my logs a see a lot of the following errors
```
{"time":"2024-12-19T15:08:46.866Z","severity":"error","message":"failed to fetch: :etimedout","metadata":{"count":100,"fetcher":"coin_balance_catchup","error_count":100}}
```

while at the same time `block_catchup` appears to be working fine.
```
{"time":"2024-12-19T15:08:54.818Z","severity":"info","message":"Index already caught up.","metadata":{"fetcher":"block_catchup","last_block_number":null,"shrunk":false,"missing_block_count":0,"first_block_number":null}}
```

Everything points to the same private node. I assume that if `block_catchup` can conclude we're not missing any blocks that the node is reachable for data queries. 

What can cause a `etimedout` for `coin_balance_catchup` though?

**Comment by vbaranov** [comment]
A relatively high batch size for a coin balance fetcher can cause a timeout issue at request. Consider decreasing the default 100 batch size via runtime env variable `INDEXER_COIN_BALANCES_BATCH_SIZE`.

**Reply by ixje** [reply]
Thanks for the suggestion. I noticed that `last_block_number` and `first_block_number` in the log message were both `null` and started thinking that that might be the reason it outputs `Index already caught up`. After some more digging it turned out to be a network reach-ability issue after all. Now `last_block_number` and `first_block_number` show numbers instead.
